src/webui/www/private/views/searchplugins.html

```diff
@@ -230,10 +232,11 @@ <h2>QBT_TR(Installed search plugins:)QBT_TR[CONTEXT=PluginSelectDlg]</h2>
             setupSearchPluginTableEvents(true);
         };
 
-        initSearchPlugins();
-
         return exports();
     })();
 
     Object.freeze(window.qBittorrent.SearchPlugins);
+    };
+
+    window.qBittorrent.SearchPlugins.init();
```

Comment:
What's the cause for this PR - are these scripts executing twice? If so, do we also need to worry about these `init` functions running multiple times?

Reply:
>What's the cause for this PR - are these scripts executing twice? If so, do we also need to worry about these init functions running multiple times?

The function definitions should only run once.
These `init` are different, they are supposed to be executed every time the file is loaded to initialize the functionality of HTML/JS.

Reply:
> Ideally, the declaration should only happen once.

Are the declarations happening more than once? Or is this change purely speculative.

Reply:
Yes, otherwise I wouldn't make this change.

Reply:
I didn't see it mentioned anywhere so seemed worthwhile to confirm. I think it would be useful to document how it's occurring as my current understanding is that this should not happen. For example, I don't see any issues reported related to this.

Reply:
>I think it would be useful to document how it's occurring as my current understanding is that this should not happen. 

I used preferences.html for testing and it is reinitialized every time when loaded. Other JS files may not be reloaded during webui lifetime but it is worth to apply the guarding pattern.

>For example, I don't see any issues reported related to this.

It is not causing visible problems so you won't see user reports it.

Reply:
> I used preferences.html for testing and it is reinitialized every time when loaded.

Is it deleted (destructed, released, freed or whatever it is called in JS) when closing the dialog? If so, then this is normal behavior for a dialog, IMO.

Reply:
>Is it deleted (destructed, released, freed or whatever it is called in JS) when closing the dialog? If so, then this is normal behavior for a dialog, IMO.

Yes, normal but it doesn't mean js engine needs to interpret the code again, it can reuse the existing one.

Reply:
> > Is it deleted (destructed, released, freed or whatever it is called in JS) when closing the dialog? If so, then this is normal behavior for a dialog, IMO.
> 
> Yes, normal but it doesn't mean js engine needs to reinterpret the full code again, it can reuse the existing one.

Damn, I'm confused. Could you explain to me the sequence of steps of its execution during the first and subsequent display of Preferences dialog (if it doesn't bother you too much)?

Reply:
>Could you explain to me the sequence of steps of its execution during the first and subsequent display of Preferences dialog (if it doesn't bother you too much)?

When Preferences dialog is loaded for the first time. `window.qBittorrent.Preferences` is `undefined` and browser will interpret/run the code and eventually `window.qBittorrent.Preferences` will become `exports();`.
When Preferences dialog is closed. `window.qBittorrent.Preferences` is retained. It won't be deallocated since `window` is a global object.
When Preferences dialog is loaded for the second time. `window.qBittorrent.Preferences` is already defined and with this PR. browser don't need to run `window.qBittorrent.Preferences = (() => {...})();` and can just use the previous object.

Reply:
> It won't be deallocated since `window` is a global object.

Is this a good idea? Why make it global instead of just deallocating it after use? After all, it may never be needed again in the current session. 

> When Preferences dialog is loaded for the second time. `window.qBittorrent.Preferences` is already defined and with this PR, browser don't need to run `window.qBittorrent.Preferences = (() => {...})();` and can just use the previous object.

Doesn't the JS engine need to parse all this code before executing it?

Reply:
@Chocobo1 
I don't pretend to anything in this topic. There are other JS-qualified people to review it.

Reply:
>Is this a good idea? Why make it global instead of just deallocating it after use? After all, it may never be needed again in the current session.

Well... in JS, every function declared at the top level will become global. So to avoid polluting the global, it was decided to put into `window.qBittorrent.*`. And unless we manually delete it, it will stay there.
Another reason for keeping the `window.qBittorrent.*` is that it may aid JS engine in performance. JS engine wouldn't use JIT to compile a function if it keeps reinitializing/changing.

>Doesn't the JS engine need to parse all this code before executing it?

Yes. To be clear, with this PR, the object `window.qBittorrent.*` won't need to regenerate/reinitialize every time when it is loaded.